cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/profile/ProfileDataset.java

```diff
@@ -110,12 +111,12 @@ public Profile getProfile(ProfileId profileId) throws NotFoundException {
    * @param includeSystem whether to include profiles in system namespace
    * @return the list of profiles which is in this namespace
    */
-  public List<Profile> getProfiles(NamespaceId namespaceId, boolean includeSystem) {
-      List<Profile> profiles = new ArrayList<>(table.list(getProfileRowKeyPrefix(namespaceId), Profile.class));
-      if (includeSystem && !namespaceId.equals(NamespaceId.SYSTEM)) {
-        profiles.addAll(table.list(getProfileRowKeyPrefix(NamespaceId.SYSTEM), Profile.class));
-      }
-      return Collections.unmodifiableList(profiles);
+  public List<ProfileMeta> getProfiles(NamespaceId namespaceId, boolean includeSystem) {
+    List<ProfileMeta> profiles = new ArrayList<>(table.list(getProfileRowKeyPrefix(namespaceId), ProfileMeta.class));
```

Comment:
This new ProfileMeta class is not needed. The caller knows the namespace and Profile contains the profile name.

Reply:
Removed.